### PR TITLE
logout link in no roles error page fix

### LINKED ISSUE #41580: Logout link in no roles and permissions error page is broken
### Apache Airflow version

main (development)

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

logout endpoint was made a POST endpoint in https://github.com/apache/airflow/pull/40145 . With this change the /logout link in error page when there are no roles and permissions is still a GET request causing the request to be invalid. 

https://github.com/apache/airflow/blob/75fb7acbaca09a040067f0a5a37637ff44eb9e14/airflow/www/templates/airflow/no_roles_permissions.html#L32-L34

### What you think should happen instead?

_No response_

### How to reproduce

1. Initialize a new database.
2. Create an admin user.
3. Login as the user without roles and permission to get to the error page.
4. Click on logout link.

### Operating System

Ubuntu

### Versions of Apache Airflow Providers

_No response_

### Deployment

Virtualenv installation

### Deployment details

_No response_

### Anything else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
closes: #41580

This PR fixes log out link in no roles and permissions error page

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by gagan-bhullar-tech** [comment]
@shahar1 can you please review

File: airflow/www/templates/airflow/no_roles_permissions.html [hunk 1]
```diff
@@ -30,7 +30,11 @@ <h1>Your user has no roles and/or permissions!</h1>
     <p>Unfortunately your user has no roles, and therefore you cannot use Airflow.</p>
     <p>Please contact your Airflow administrator
       (<a href="https://airflow.apache.org/docs/apache-airflow/stable/security/webserver.html#web-authentication">authentication</a>
-      may be misconfigured) or <a href="{{ logout_url }}">log out</a> to try again.</p>
+      may be misconfigured) or 
+      <form method="POST" action="{{logout_url}}" id="logout-form">
+        <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
+        <a href="javascript:{}" onclick="document.getElementById('logout-form').submit();">log out</a> to try again.
+      </form>
```
Comment (rohitnandi12) [inline airflow/www/templates/airflow/no_roles_permissions.html:37]: There is a missing `</p>` closing tag which opens at line 31 `<p>`

**Review by rohitnandi12 — CHANGES_REQUESTED** [review]
missing Paragraph closing tag

File: airflow/www/templates/airflow/no_roles_permissions.html (hunk 1 above)
Comment (gagan-bhullar-tech) [reply airflow/www/templates/airflow/no_roles_permissions.html:37]: @rohitnandi12 thanks for pointing it out. Updated PR.

**Review by shahar1 — APPROVED** [review]

**Comment by shahar1** [comment]
Excellent! Thank you

**Pull request merged**